stack2nix/Stack2nix.hs

```diff
@@ -140,16 +151,30 @@ packages2nix args (Stack _ _ pkgs) =
           return $ \sha256 ->
             forM cabalFiles $ \cabalFile -> do
             let pkg = cabalFilePkgName cabalFile
-                nix = ".stack.nix" </> pkg <.> "nix"
-                nixFile = outputPath args </> nix
+                nix = pkg <.> "nix"
+                nixFile = argOutputDir args </> nix
                 subdir' = if subdir == "." then Nothing
                           else Just subdir
                 src = Just $ C2N.Git url rev (Just sha256) subdir'
             createDirectoryIfMissing True (takeDirectory nixFile)
             writeDoc nixFile =<<
               prettyNix <$> cabal2nix src cabalFile
-            liftIO $ appendCache cacheFile url rev subdir sha256 pkg nix
+            liftIO $ appendCache (argCacheFile args) url rev subdir sha256 pkg nix
             return $ fromString pkg $= mkPath False nix
 
-cacheFile :: FilePath
-cacheFile = ".stack-to-nix.cache"
+defaultNixContents :: String
```

Comment:
No `hnix` usage to generate the ast? Should we just have a `tmpl` data folder, and copy the default template from there?

Reply:
Yeah I think hnix will be the best way to generate it. `String` is just for the draft version.